src/sharedarraybuffer-metadata.h

```diff
@@ -0,0 +1,66 @@
+#ifndef SRC_SHAREDARRAYBUFFER_METADATA_H_
+#define SRC_SHAREDARRAYBUFFER_METADATA_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "node.h"
+#include <memory>
+
+namespace node {
+namespace worker {
+
+class SharedArrayBufferMetadata;
+
+// This is an object associated with a SharedArrayBuffer, which keeps track
+// of a cross-thread reference count. Once a SharedArrayBuffer is transferred
+// for the first time (or is attempted to be transferred), one of these objects
+// is created, and the SharedArrayBuffer is moved from internalized mode into
+// externalized mode (i.e. the JS engine no longer frees the memory on its own).
+//
+// This will always be referred to using a std::shared_ptr, since it keeps
+// a reference count and is guaranteed to be thread-safe.
+typedef std::shared_ptr<SharedArrayBufferMetadata>
+    SharedArrayBufferMetadataReference;
+
+class SharedArrayBufferMetadata
+    : public std::enable_shared_from_this<SharedArrayBufferMetadata> {
+ public:
+  static SharedArrayBufferMetadataReference ForIncomingSharedArrayBuffer(
+      Environment* env, v8::Local<v8::Context> context,
+      v8::Local<v8::SharedArrayBuffer> source);
+  ~SharedArrayBufferMetadata();
+
+  // Create a SharedArrayBuffer object for a specific Environment and Context.
+  // The created SharedArrayBuffer will be in externalized mode and has
+  // a hidden object attached to it, during whose lifetime the reference
+  // count if increased by 1.
```

Comment:
s/if/is/

Reply:
Actually the whole wording of this is a bit weird. Not sure what would be a better way to explain that. 🤔

Reply:
> Not sure what would be a better way to explain that. :thinking:

Hm, yeah… was this comment helpful in explaining what’s happening?

Reply:
It makes sense for the most part. The wording of `during whose lifetime the reference count if increased by 1` was a bit hard for me to parse though. I suspect someone even less familiar with threading stuff would be even more confused.